src/linear_algebra/basis.lean

```diff
@@ -397,6 +397,11 @@ by rw [b.constr_def S f, linear_map.range_comp, linear_map.range_comp, linear_eq
        ← finsupp.supported_univ, finsupp.lmap_domain_supported, ←set.image_univ,
        ← finsupp.span_eq_map_total, set.image_id]
 
+@[simp]
+lemma constr_comp [module S M] [smul_comm_class R S M] (f : M →ₗ[R] M) (v : ι → M) :
```

Comment:
What's the difference between `M` and `M'` here? I think `M'` already has the assumptions you need.

Reply:
```suggestion
lemma constr_comp (f : M' →ₗ[R] M') (v : ι → M') :
```